platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py

```diff
@@ -25,6 +26,15 @@
 #
 EEPROM_SYMLINK = "/var/run/hw-management/eeprom/vpd_info"
 
+platform_name = get_platform()
+if 'simx' in platform_name:
+    platform_path = get_path_to_platform_dir()
+
+    if not os.path.exists(os.path.dirname(EEPROM_SYMLINK)):
+        os.makedirs(os.path.dirname(EEPROM_SYMLINK))
+    if not os.path.exists(EEPROM_SYMLINK):
+        subprocess.check_call(['/usr/bin/xxd', '-r', '-p', 'syseeprom.hex', EEPROM_SYMLINK], cwd=platform_path)
```

Comment:
@tomer-israel why do we need to always check both?

Example:
```cpp
if not os.path.exists(EEPROM_SYMLINK):
    if not os.path.exists(os.path.dirname(EEPROM_SYMLINK)):
        os.makedirs(os.path.dirname(EEPROM_SYMLINK))

    subprocess.check_call(['/usr/bin/xxd', '-r', '-p', 'syseeprom.hex', EEPROM_SYMLINK], cwd=platform_path)
```

Reply:
changed it as you suggested